File: javascript/QuickSort.js
let n = prompt("Digite o tamanho do vetor: ");
const vetor = [];

for(let i = 0; i < n; i++){
  let num = prompt("Digite o valor da "+ i + "º pos.");
  num = parseInt(num)
  vetor.push(num)
}

for (let j = 0; j < n - 1; j++) { //Simulando a recursividade
  for (let i = 0; i < n - 1 - j; i++) {
    if (vetor[i] > vetor[i + 1]) {
      let aux = vetor[i];
      vetor[i] = vetor[i + 1];
      vetor[i + 1] = aux;
    }
  }
}
console.log(vetor)

File: java/QuickSort.java
import java.util.Scanner;

public class QuickSort{
  Scanner prompt = new Scanner(System.in);
  
  public void DoQuickSort(){
    
    System.out.println("Digite o tamanho do vetor: ");
    int n = prompt.nextInt();
    int[] vetor = new int[n];
    for(int i = 0; i < n; i++){
      System.out.println("Digite o valor da "+ i + "º pos.");
      vetor[i] = prompt.nextInt();
    }

    for (int j = 0; j < n - 1; j++) {
      for (int i = 0; i < n - 1 - j; i++) {
        if (vetor[i] > vetor[i + 1]) {
          int aux = vetor[i];
          vetor[i] = vetor[i + 1];
          vetor[i + 1] = aux;
        }
      }
    }

    System.out.println("After QuickSort");
    for (int i = 0; i < n; i++){
      System.out.println(vetor[i]);
    }
  }
}
